### PR TITLE
[SHRINKRES-342] Migrate to JUnit5

### PR DESCRIPTION
**Pull request by xjusko**
https://issues.redhat.com/browse/SHRINKRES-342

### CODE REVIEW

File: maven/impl-maven-archive/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/archive/importer/ConfiguredMavenImporterTestCase.java [hunk 1]
```diff
@@ -31,46 +31,42 @@
 import org.jboss.shrinkwrap.resolver.api.NoResolvedResultException;
 import org.jboss.shrinkwrap.resolver.api.maven.archive.importer.MavenImporter;
 import org.jboss.shrinkwrap.resolver.impl.maven.archive.util.TestFileUtil;
-import org.junit.After;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.BeforeClass;
-import org.junit.Test;
+import org.junit.jupiter.api.*;
```
Comment (petrberan) [inline maven/impl-maven-archive/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/archive/importer/ConfiguredMavenImporterTestCase.java:34]: No wildcard imports please

Comment (xjusko) [reply maven/impl-maven-archive/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/archive/importer/ConfiguredMavenImporterTestCase.java]: I have missed that. All the occurrences should be fixed.

File: maven/impl-maven-archive/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/archive/util/JarGeneratorTestCase.java [hunk 2]
```diff
@@ -42,24 +42,19 @@
...
  * A utility to generate artifact jars
  *
  * @author <a href="mailto:kpiwko@redhat.com>Karel Piwko</a>
  *
  */
-@RunWith(Parameterized.class)
-public class JarGeneratorTestCase {
-    private final String name;
-    private final Class<?>[] content;
+class JarGeneratorTestCase {
```
Comment (petrberan) [inline maven/impl-maven-archive/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/archive/util/JarGeneratorTestCase.java:54]: This is misleading, there are no tests, thus this can't be *TestCase. Same with WarGeneratorTestCase

File: maven/impl-maven/src/test/java/org/jboss/shrinkwrap/resolver/api/maven/coordinate/MavenDependenciesTestCase.java [hunk 3]
```diff
@@ -1,19 +1,3 @@
-/*
```
Comment (petrberan) [inline maven/impl-maven/src/test/java/org/jboss/shrinkwrap/resolver/api/maven/coordinate/MavenDependenciesTestCase.java:1]: Why was this removed?

File: maven/impl-maven/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/bootstrap/SystemPropertyPrecedenceTestCase.java [hunk 4]
```diff
@@ -79,24 +79,21 @@ public void overrideGlobalSettings() {
...
     }
 
-    @Test(expected = NoResolvedResultException.class)
-    public void overrideOfflineFlag() {
-
-        try {
-            System.setProperty(MavenSettingsBuilder.ALT_MAVEN_OFFLINE, "true");
+    @Test
+    void overrideOfflineFlag() {
+        System.setProperty(MavenSettingsBuilder.ALT_MAVEN_OFFLINE, "true");
```
Comment (petrberan) [inline maven/impl-maven/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/bootstrap/SystemPropertyPrecedenceTestCase.java:84]: The problem I see in this change is that in case of a test failure the property is still set and might affect other tests either in this testsuite or somewhere else (as it's a system property). I'd propose to change this testsuite in this PR and create a proper cleanup method AfterEach

File: maven/impl-maven/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/integration/OfflineRepositoryTestCase.java [hunk 5]
```diff
@@ -139,28 +142,30 @@ public void offlineProgrammaticallyPomBased() throws IOException {
...
+    void offlineBySysProp() {
         System.setProperty(MavenSettingsBuilder.ALT_MAVEN_OFFLINE, "true");
         try {
-            Maven.configureResolver().fromFile("target/settings/profiles/settings-jetty.xml")
-                    .resolve("junit:junit:3.8.2").withTransitivity().as(File.class);
-            Assert.fail("Artifact junit:junit:3.8.2 should not be present in local repository");
+            Assertions.assertThrows(NoResolvedResultException.class, () -> {
+                Maven.configureResolver().fromFile("target/settings/profiles/settings-jetty.xml")
+                        .resolve("junit:junit:3.8.2").withTransitivity().as(File.class);
+                Assertions.fail("Artifact junit:junit:3.8.2 should not be present in local repository");
```
Comment (petrberan) [inline maven/impl-maven/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/integration/OfflineRepositoryTestCase.java:152]: Isn't this Assertions.fail redundant?

File: pom.xml [hunk 6]
```diff
@@ -143,11 +143,16 @@
...
-                <artifactId>junit</artifactId>
+                <groupId>org.junit.jupiter</groupId>
+                <artifactId>junit-jupiter</artifactId>
                 <version>${version.junit}</version>
                 <scope>test</scope>
             </dependency>
+            <dependency>
+                <groupId>org.hamcrest</groupId>
+                <artifactId>hamcrest</artifactId>
+                <version>2.2</version>
```
Comment (petrberan) [inline pom.xml:154]: I believe Hamcrest released a new version 3.0, can that be used?

**Comment by petrberan** [comment]
Good job @xjusko , just a couple of things I noticed and one rebase and I believe we can merge this

File: maven/impl-maven/src/test/java/org/jboss/shrinkwrap/resolver/api/maven/coordinate/MavenDependenciesTestCase.java (hunk 3 above)
Comment (xjusko) [reply maven/impl-maven/src/test/java/org/jboss/shrinkwrap/resolver/api/maven/coordinate/MavenDependenciesTestCase.java:1]: This was not intentional, probably some random input from my keyboard... Thanks for noticing it.

**Review by petrberan — APPROVED** [review]
LGTM, thx @xjusko
